internal/agent/device/dependency/dependency.go

```diff
@@ -207,7 +223,15 @@ func (m *prefetchManager) processTarget(ctx context.Context, target string) {
 		}
 		if err := m.pull(ctx, target, task); err != nil {
 			if errors.IsRetryable(err) {
-				m.log.Debugf("Retryable error during prefetch: %v; retrying after %s", err, queueNextItemAfter)
+				// cleanup file system from partial layer pulls
+				cleanupCtx, cleanupCancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
Why a background context here instead of the main ctx? If it's a retryable error we know the context hasn't been canceled.

Reply:
I believe that context was canceled

Reply:
Well context cancelled is retryable

Reply:
```
flightctl-agent[1659]: time="2025-08-08T21:27:54.541903Z" level=warning msg="A retriable error occurred: pull image: context deadline exceeded: code: 124: context deadline exceeded" file="agent/client/podman.go:651"
```